opal/mca/common/ucx/common_ucx_wpool.c

```diff
@@ -630,21 +675,24 @@ _tlocal_add_ctx_rec(opal_common_ucx_ctx_t *ctx)
     return NULL;
 }
 
-static int _tlocal_ctx_connect(_ctx_record_t *ctx_rec, int target)
+static int _tlocal_ctx_connect(opal_common_ucx_wpmem_t *wpmem, _ctx_record_t *ctx_rec, int target)
 {
     ucp_ep_params_t ep_params;
     opal_common_ucx_winfo_t *winfo = ctx_rec->winfo;
-    opal_common_ucx_ctx_t *gctx = ctx_rec->gctx;
+    opal_common_ucx_ctx_t   *gctx  = ctx_rec->gctx;
+    opal_common_ucx_wpool_t *wpool = gctx->wpool;
     ucs_status_t status;
-    int displ;
+    unsigned int proc_vpid;
 
     memset(&ep_params, 0, sizeof(ucp_ep_params_t));
     ep_params.field_mask = UCP_EP_PARAM_FIELD_REMOTE_ADDRESS;
 
     opal_mutex_lock(&winfo->mutex);
-    displ = gctx->recv_worker_displs[target];
-    ep_params.address = (ucp_address_t *)&(gctx->recv_worker_addrs[displ]);
-    status = ucp_ep_create(winfo->worker, &ep_params, &winfo->endpoints[target]);
+    
+    proc_vpid = gctx->get_proc_vpid_func(wpmem->metadata, target);
+
+    ep_params.address = (ucp_address_t *)wpool->recv_worker_addrs[proc_vpid];
```

Comment:
1. Move this into the wpool scope (an access function)
2. Explain in the comment to this function why we don't need lock.